modules/payment/commerce_payment.routing.yml

```diff
@@ -62,3 +62,25 @@ entity.commerce_payment_method.collection:
     parameters:
       user:
         type: entity:user
+
+# @todo only generate if checkout is enabled.
```

Comment:
If I remember correctly, a route can have a requirement on an enabled module, which would solve this.